sympy/ntheory/factor_.py

```diff
@@ -1419,6 +1419,53 @@ def divisor_count(n, modulus=1):
     return Mul(*[v + 1 for k, v in factorint(n).items() if k > 1])
 
 
+def proper_divisors(n, generator=False):
+    """
+    Return all divisors of n except 1 and n, sorted by default.
+    If generator is ``True`` an unordered generator is returned.
+
+    Examples
+    ========
+    
+    >>> from sympy import proper_divisors, proper_divisor_count
+    >>> proper_divisors(24)
+    [2, 3, 4, 6, 8, 12]
+    >>> proper_divisor_count(24)
+    6
+    >>> list(proper_divisors(120, generator=True))
+    [2, 4, 8, 3, 6, 12, 24, 5, 10, 20, 40, 15, 30, 60]
+
+    See Also
+    ========
+    
+    factorint, divisors, proper_divisor_count
+    
+    """
+    for i in divisors(n, generator):
```

Comment:
This does not not give a generator if `generator=False`. You will need to have a separate internal function that is a generator that gets returned when `generator=True`.